components/portals/portals.js

```diff
@@ -0,0 +1,135 @@
+var EventEmitter = require('node-event-emitter');
+var React = require('react');
+var warning = require('react/lib/warning');
+
+var types = React.PropTypes;
+
+var destinationPortals = {};
+var emitter = new EventEmitter();
+
+function createRoot(reactElement) {
+  var destination = document.createElement('div');
+  reactElement.getDOMNode().appendChild(destination);
+  return destination;
+}
+
+function reset() {
+  emitter.removeAllListeners();
+  destinationPortals = {};
+}
+
+/**
+ * @component PortalSource
+ * @description HTML Rendered to PortalSource will be transported to PortalDestination
+ *
+ * @property name {String} Name for linking source and destination portals
```

Comment:
Could we rename this property to `dest`? That way it seems more like the source is pointing to the dest.